pulsar-io/mongo/src/main/java/org/apache/pulsar/io/mongodb/MongoSource.java

```diff
@@ -133,9 +134,12 @@ public void onNext(ChangeStreamDocument<Document> doc) {
                     recordValue.put("ns", doc.getNamespace());
                     recordValue.put("operation", doc.getOperationType());
 
-                    consume(new DocRecord(
-                            Optional.of(doc.getDocumentKey().toJson()),
-                            mapper.writeValueAsString(recordValue).getBytes(StandardCharsets.UTF_8)));
+                    BsonDocument key = doc.getDocumentKey();
+                    if (key != null) {
+                        consume(new DocRecord(
+                                Optional.of(key.toJson()),
+                                mapper.writeValueAsString(recordValue).getBytes(StandardCharsets.UTF_8)));
+                    }
```

Comment:
Should an exception be thrown if `doc.getDocumentKey()` returns null?

Reply:
Or in this way?

```java
BsonDocument key = doc.getDocumentKey();
consume(new DocRecord(
        key != null ? Optional.of(key.toJson()) : Optional.empty(),
        mapper.writeValueAsString(recordValue).getBytes(StandardCharsets.UTF_8)));
```

Reply:
@shink I think we don't need to throw exception. We also can
```java
Optional.ofNullable(doc.getDocumentKey()).map(BsonDocument::toJson),
```

Reply:
@Shoothzj Good idea. Done.